File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr, int k) {
        int[] ret = new int[k];
        Arrays.fill(ret, -1);
        // System.out.println(Arrays.toString(arr));
        Map<Integer, Integer> mp = new HashMap<>();
        for(int i = 0, j = 0; i < arr.length; i++){
            if(j == k) break;
            if(mp.containsKey(arr[i])) continue;
            mp.put(arr[i], 1);
            ret[j++] = arr[i];
        }
        // System.out.println(Arrays.toSTrin);
        
        return ret;
    }
}

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.py
def solution(arr, k):
    ret, st = [], set()
    for x in arr:
        if x not in st:
            ret.append(x)
            st.add(x)
        if len(ret) == k: break
        
    while len(ret) < k:
        ret.append(-1)
    return ret
    
    